Document `steps` configuration for all possible runbook actions (#81)
firehydrant/terraform-provider-firehydrant: docs/guides/runbooks_steps_pagerduty.md

subcategory: "Runbooks"
---

# PagerDuty

~> **Note** You must have the PagerDuty integration installed in FireHydrant
for any PagerDuty runbook steps to work properly.

The FireHydrant PagerDuty integration allows FireHydrant users to link incidents
in PagerDuty to incidents in FireHydrant.

### Available Steps

* [Create PagerDuty Incident](#create-pagerduty-incident)

## Create PagerDuty Incident

The [PagerDuty **Create PagerDuty Incident** step](https://support.firehydrant.com/hc/en-us/articles/360057722212-Starting-a-PagerDuty-Incident)
allows FireHydrant users to create a new PagerDuty incident from a FireHydrant incident.

### Create PagerDuty Incident - Example Usage

Basic usage:
```hcl
data "firehydrant_runbook_action" "pagerduty_create_new_incident" {
  integration_slug = "pager_duty"
  slug             = "create_new_pager_duty_incident"
  type             = "incident"
}

resource "firehydrant_runbook" "pagerduty_create_new_incident_runbook" {
  name = "pagerduty-create-new-incident-runbook"

  steps {
    name      = "Create PagerDuty Incident"
    action_id = data.firehydrant_runbook_action.pagerduty_create_new_incident.id

    config = jsonencode({

>       incident_creator = {
>         label = ""
>         value = ""
>       }
>       service = {
>         label = ""
>         value = ""
>       }
>       escalation_policy = {
>         label = ""
>         value = ""
>       }

Author: ```suggestion
      incident_creator = {
        label = "user@example.com"
        value = "user@example.com"
      }
      service_id = {
        label = "Registration Service"
        value = "xxxxxxx"
      }
      escalation_policy_id = {
        label = "Registration Team Escalation Policy"
        value = "xxxxxxx"
      }
```

      incident_title   = "{{incident.severity}} Incident: {{incident.name}}"
      incident_details = "There is a {{incident.severity}} incident.\n\nTriggered by {{author.name}} from FireHydrant. For more information see {{incident.incident_url}}."
    })
    
    automatic = true
  }
}
```

### Create PagerDuty Incident - Steps Argument Reference

* `action_id` - (Required) The ID of the runbook action for the step.
* `config` - (Required) JSON string representing the configuration settings for the step.
  Use [Terraform's jsonencode](https://www.terraform.io/language/functions/jsonencode)
  function so that [Terraform can guarantee valid JSON syntax](https://www.terraform.io/language/expressions/strings#generating-json-or-yaml).
* `name` - (Required) The name of the step.
* `automatic` - (Optional) Whether this step should automatically execute.

The `config` block supports:

* `escalation_policy_id` - (Required) The PagerDuty escalation policy for determining the users to be alerted.
* `incident_creator` - (Required) The PagerDuty user to use for creating the incident.
* `service_id` - (Required) The PagerDuty service to include in the incident.

> * `incident_details` - (Optional) A description of the incident for PagerDuty.
>   This field supports [FireHydrant's template variables](https://support.firehydrant.com/hc/en-us/articles/4409136426004-Using-template-variables-in-Runbooks)
>   so you can automatically include details such as when the incident started, the incident summary, severity, roles involved, and much more.
> * `incident_title` - (Optional) The title of the PagerDuty incident.
>   This field supports [FireHydrant's template variables](https://support.firehydrant.com/hc/en-us/articles/4409136426004-Using-template-variables-in-Runbooks)
>   so you can automatically include details such as when the incident started, the incident summary, severity, roles involved, and much more.

Author: these appear to all be required, requests fail without them
```suggestion
* `incident_details` - (Required) A description of the incident for PagerDuty.
  This field supports [FireHydrant's template variables](https://support.firehydrant.com/hc/en-us/articles/4409136426004-Using-template-variables-in-Runbooks)
  so you can automatically include details such as when the incident started, the incident summary, severity, roles involved, and much more.
* `incident_title` - (Required) The title of the PagerDuty incident.
  This field supports [FireHydrant's template variables](https://support.firehydrant.com/hc/en-us/articles/4409136426004-Using-template-variables-in-Runbooks)
  so you can automatically include details such as when the incident started, the incident summary, severity, roles involved, and much more.
```

The `escalation_policy_id` block supports:

> * `label` - (Required) The name of the PagerDuty team.
> * `value` - (Required) The ID of the PagerDuty team.

Author: ```suggestion
* `label` - (Required) The name of the PagerDuty escalation policy.
* `value` - (Required) The ID of the PagerDuty escalation policy.
```

The `incident_creator` block supports:

> * `label` - (Required) The name of the PagerDuty team.
> * `value` - (Required) The ID of the PagerDuty team.

Author: ```suggestion
* `label` - (Required) The email of the PagerDuty user to set as the incident creator. This value should match the `value` attribute.
* `value` - (Required) The email of the PagerDuty user to set as the incident creator. This value should match the `label` attribute.
```

The `service_id` block supports:

> * `label` - (Required) The name of the PagerDuty team.
> * `value` - (Required) The ID of the PagerDuty team.

Author: ```suggestion
* `label` - (Required) The name of the PagerDuty service.
* `value` - (Required) The ID of the PagerDuty service.
```
